.github/workflows/yocto-build-deploy.yml

```diff
@@ -0,0 +1,1174 @@
+name: "Yocto Build-Test-Deploy"
+
+on:
+  workflow_call:
+    secrets:
+      # API key names should be descriptive telling us what company, what for, where from, what's the key.
+      # Let's say, XCompany wants to have test, then keys could be: XCOMPANY_AUTOKIT_BM_ACCESS_TOKEN
+      # TODO: Use Github CLI or API to fetch the correct key based on environment, instead of having separate secrets for each env
+      AUTOKIT_BALENACLOUD_API_KEY:
+        description: balena API key for Leviathan testing from BalenaCloud
+        required: false
+      AUTOKIT_BALENAMACHINE_API_KEY:
```

Comment:
This secret will be deprecated once we deploy the `BALENA_API_KEY` secret to each environment.